### PR TITLE
OPHJOD-484: Add OsaamisSuosittelija to Tools/Competences page

### PR DESCRIPTION
**Pull request by ketsappi**
## Description

Add OsaamisSuosittelija component to use on the Tools/Competences page.

Able to show competences based on search and select those; not saving selected ones anywhere at the moment.

## Related JIRA ticket

https://jira.eduuni.fi/browse/OPHJOD-484

### CODE REVIEW

File: src/routes/Tool/Competences.tsx [hunk 1]
```diff
@@ -159,17 +129,26 @@ const Competences = () => {
...
+          {!isLoading ? (
+            <FormProvider {...methods}>
+              <Form>
+                <InputField
+                  label={t('work-history.job-duties')}
+                  value={tyotehtava}
+                  onChange={(event: ChangeEvent<HTMLInputElement>) => setTyotehtava(event.target.value)}
+                  placeholder="Lorem ipsum dolor sit amet"
+                  help="Help text"
+                />
```
Comment (sauanto) [inline src/routes/Tool/Competences.tsx:141]:
```suggestion
                <div className="mb-6">
                  <InputField
                    label={t('work-history.job-duties')}
                    value={tyotehtava}
                    onChange={(event: ChangeEvent<HTMLInputElement>) => setTyotehtava(event.target.value)}
                    placeholder="Lorem ipsum dolor sit amet"
                    help="Help text"
                  />
                </div>
```

File: src/routes/Tool/Competences.tsx [hunk 2]
```diff
@@ -146,6 +92,30 @@ const Competences = () => {
...
+    defaultValues: async () => {
+      return Promise.resolve({
+        osaamiset: [],
+      });
+    },
+  });
+
+  const { isLoading } = useFormState({
+    control: methods.control,
+  });
```
Comment (sauanto) [inline src/routes/Tool/Competences.tsx:117]: Is React Form Hook a little redundant at this part?

**Review by sauanto — CHANGES_REQUESTED** [review]

File: src/routes/Tool/Competences.tsx (hunk 2 above)
Comment (ketsappi) [reply src/routes/Tool/Competences.tsx:117]: I thought that when submitting the selected competences later down the road it is done with form's `onSubmit`

File: src/routes/Tool/Competences.tsx (hunk 1 above)
Comment (ketsappi) [reply src/routes/Tool/Competences.tsx:141]: It seems that in the design the margin I missed it is `16px`, so it should instead be `mb-5`

File: src/routes/Tool/Competences.tsx (hunk 2 above)
Comment (ketsappi) [reply src/routes/Tool/Competences.tsx]: This should be now changed to a lot simpler.

**Review by sauanto — APPROVED** [review]

**Pull request merged**